packages/gauche/ext-threads-threads.c.patch

```diff
@@ -0,0 +1,12 @@
+--- a/ext/threads/threads.c
++++ b/ext/threads/threads.c
+@@ -454,7 +454,9 @@
+                 /* Last resort */
+                 thread_cleanup_inner(target);
+ #if defined(GAUCHE_USE_PTHREADS)
++#ifndef __ANDROID__
+                 pthread_cancel(target->thread);
```

Comment:
This change possibly breaks threading. `pthread_cancel` usually should be replaced by equivalent implementation through atomic flags.

Reply:
Now `pthread_kill` is used instead of `pthread_cancel`. This is a last resort which does not need to be as graceful as using atomic flags. Thanks!

Reply:
`pthread_kill(target->thread, 0);` is a no-op and was source of serious bugs in some our packages previously. Check issues for package `calcurse`.

However if you do not observe any problems (package works fine), you can remove this line.

Reply:
At least it passes bundled tests when `pthread_cancel` is removed. But the safety is still questionable.

An upstream issue is now open: https://github.com/shirok/Gauche/issues/790

Reply:
Approved upstream.